src/QueryBuilder/QueryViewer/QueryViewer.js

```diff
@@ -0,0 +1,124 @@
+import React, { memo, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import QueryLoader from '../QueryLoader/QueryLoader';
+import { getTableMetadata } from '../helpers';
+
+const QueryViewer = ({
+  showPagination = true,
+  contentData = [],
```

Comment:
I think it should not be a part of components. For every application we can do that in different way (some using axios, some fetch, some react-query etc). I'm just inspecting 2 things from API, entityType + contentData

Reply:
It Lists app for example there are 2 separate API's for getting entityType and contentData, but it other places it should be combined in 1 if there is no pagination (case with 100 items preview for example)

Reply:
From outside you should provide ONLY initial value for the `limit` and `offsset` - if you want to overwrite it, but not the limit and offset state. I agree that state from props - is anti-pattern, but in case of initial value which not be changed across the renders, it should be fine. I saw this pattern in MaterialReactTable, for example - https://www.material-react-table.com/docs/guides/pagination#customize-pagination-components
You can do the same actually, add initial state of limit and offset and use `onPageChanged` in parent.

Reply:
Maybe it's reasonable to change props names to initialLimit and initialOffset for avoid confusion

Reply:
my point is not only about limit and offset, it's about approach in general, so that's why I would propose to discuss contract because this viewer can work in different ways and with different data source

with current implementation it's hard to support

Reply:
Replaced with async data sources